src/unit_tests/agentlessd/test_agentlessd.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+
+#include "headers/shared.h"
+#include "agentlessd/agentlessd.h"
+#include "../wrappers/common.h"
+#include "../wrappers/wazuh/shared/debug_op_wrappers.h"
+#include "../wrappers/wazuh/shared/mq_op_wrappers.h"
+#include "../wrappers/libc/stdio_wrappers.h"
+#include "../wrappers/posix/unistd_wrappers.h"
+#include "../wrappers/libc/stdio_wrappers.h"
+
+// internal use (static) not in agentlessd/agentlessd.h
+int save_agentless_entry(const char *host, const char *script, const char *agttype);
+int send_intcheck_msg(const char *script, const char *host, const char *msg);
+int send_log_msg(const char *script, const char *host, const char *msg);
+int gen_diff_alert(const char *host, const char *script, time_t alert_diff_time);
+#define BUFFER_SIZE OS_MAXSTR - (OS_LOG_HEADER * 2)
+#define STR_MORE_CHANGES "More changes..."
+
+static int test_setup(void **state) {
+    (void) state;
+    test_mode = 1;
+    return OS_SUCCESS;
+}
+
+static int test_teardown(void **state) {
+    (void) state;
+    test_mode = 0;
+    return OS_SUCCESS;
+}
+
+#define HOST "host"
+#define SCRIPT "script"
+#define AGTTYPE "agttype"
+#define MSG "msg"
+#define STR_HELPER(s) #s
+#define STR(s) STR_HELPER(s)
+#define ADT 1667392718
+
+void test_gen_diff_alert(void **state) {
+    (void) state;
+
+    expect_string(__wrap_fopen, path, DIFF_DIR "/" HOST "->" SCRIPT "/diff." STR(ADT));
+    expect_string(__wrap_fopen, mode, "r");
+    will_return(__wrap_fopen, 1);
+
+    expect_fread(DIFF_DIR "/" HOST "->" SCRIPT "/diff." STR(ADT), BUFFER_SIZE);
+
+    expect_SendMSG_call("ossec: agentless: Change detected:\n" STR_MORE_CHANGES, 
+                "(" SCRIPT ") " HOST "->wazuh-agentlessd",
+                LOCALFILE_MQ, 0);
+
+    // save_agentless_entry() call
+    expect_string(__wrap_fopen, path, AGENTLESS_ENTRYDIR "/(" SCRIPT ") " HOST);
+    expect_string(__wrap_fopen, mode, "w");
+    will_return(__wrap_fopen, 2);
+
+    expect_any(__wrap_fprintf, __stream);
+    expect_string(__wrap_fprintf, formatted_msg, "type: diff\n");
+    will_return(__wrap_fprintf, 0);
+
+    expect_fclose((FILE *)2, 0);
+    // save_agentless_entry() call
+
+    expect_fclose((FILE *)1, 0);
+
+    gen_diff_alert(HOST, SCRIPT, ADT);
```

Comment:
this function always return 0, even out with error or not.

Reply:
hwo to proced in these cases?

Reply:
Well, that seems to be a problem of the method and we won't fix that in this PR. I didn't noticed that.

But we should check the returned value anyway, because if the tested function changes in the future, we'll force the developer to check and update the UT too.